pkg/security/ImageScanService.go

```diff
@@ -593,33 +594,46 @@ func (impl *ImageScanServiceImpl) getImageScanOutputObjectsV1(stepOutput []byte,
 
 func (impl *ImageScanServiceImpl) getImageScanOutputObjectsV2(stepOutput []byte, resultDescriptorTemplate string) ([]*bean.ImageScanOutputObject, error) {
 	var vulnerabilities []*bean.ImageScanOutputObject
-	var mappings []bean.Mapping
+	var mappings []map[string]interface{}
 	err := json.Unmarshal([]byte(resultDescriptorTemplate), &mappings)
 	if err != nil {
 		impl.Logger.Errorw("error in un-marshaling result descriptor template", "err", err, "resultDescriptorTemplate", resultDescriptorTemplate)
 		return nil, err
 	}
-	var processArray func(mapping bean.Mapping, value gjson.Result)
-	processArray = func(mapping bean.Mapping, value gjson.Result) {
+	var processArray func(mapping map[string]interface{}, value gjson.Result)
+	processArray = func(mapping map[string]interface{}, value gjson.Result) {
+		vulnerabilitiesPath := mapping[bean.MappingKeyPathToVulnerabilitiesArray].(string)
+		vulnerabilityDataKeyPathsMap := mapping[bean.MappingKeyPathToVulnerabilityDataKeys].(map[string]interface{})
+		resultDataKeyPathsMap := mapping[bean.MappingKeyPathToResultDataKeys].(map[string]interface{})
```

Comment:
why isn't `resultDataKeyPathsMap` data type is map[string]string ??

Reply:
since we decided to group the `vulnerability keys` and `results data keys` into separate structs in the mapping json